solutions/09/script.js

```diff
@@ -1,30 +1,63 @@
-function validateCardNumber(number) {
-    var regex = new RegExp("^[0-9]{16}$");
-    if (!regex.test(number))
-        return false;
+function validateCardNumber(value) {
+    // Accept only digits, dashes or spaces
+	if (/[^0-9-\s]+/.test(value)) return false;
 
-    return luhnCheck(number);
+	// The Luhn Algorithm. It's so pretty.
+	let nCheck = 0, bEven = false;
+	value = value.replace(/\D/g, "");
+
+	for (let n = value.length - 1; n >= 0; n--) {
+		let cDigit = value.charAt(n),
+			  nDigit = parseInt(cDigit, 10);
+
+		if (bEven && (nDigit *= 2) > 9) nDigit -= 9;
+
+		nCheck += nDigit;
+		bEven = !bEven;
+	}
+	return (nCheck % 10) == 0;
 }
 
-function luhnCheck(val) {
-    var sum = 0;
-    for (var i = 0; i < val.length; i++) {
-        var intVal = parseInt(val.substr(i, 1));
-        if (i % 2 == 0) {
-            intVal *= 2;
-            if (intVal > 9) {
-                intVal = 1 + (intVal % 10);
-            }
+function whatCard(value) {
+    const icon = document.querySelector('.icon');
```

Comment:
const icon = document.querySelector('.icon');
better get it once, not every time when you run function.

Reply:
What about old class?

Reply:
Good point with const icon. What do you mean by 'old class'

Reply:
```
icon.className = "";
            icon.classList.add('icon');
            icon.classList.add('fa');
```

Reply:
Changed to `icon.className = "icon fa";`